chapters/en/slides/module3/module3_04.md

```diff
@@ -3,141 +3,188 @@ type: slides
 
-To snapshot the changes with an associated commit message, you must put a message in the text box at the bottom of the Git pane and click on the blue “Commit” button (see image). It is highly recommended to write useful and meaningful messages about what was changed. These commit messages, and the datetime stamp for a given commit, are the primary means to navigate through the project’s history in the event that you need to view or retrieve a past version of a file, or revert your project to an earlier state. When you click the “Commit” button for the first time, you will be prompted to enter your name and email. This only needs to be done once for each machine you use Git on.
+These commit messages, and the time stamp for a given commit, are the primary means to navigate through the project’s history in the event that you need to view or retrieve a past version of a file, or revert your project to an earlier state.
+
+When you click the "Commit" button for the very first time, you will be prompted to enter your name and email. This only needs to be done once for each machine you use Git on, and is added as author information to the commit message.
 
 ---
 
 ## JupyterLab: Making the commit
 
 <center>
 
-<img src='/module3/vc-commits-jupyter-4_2.png' width="90%" alt="404 image"/>
+<img src='/module3/vc-commits-jupyter-4_2.png' width="80%" alt="404 image"/>
+
+</center>
+
+Notes:
 
-<center/>
+**Step 2 continued**
 
-Notes: You need to include a title for your commit. If you wantm you can also include an optional description for the commit.
+When committing changes, you need to include a title/summary for each message (the upper text box in JupyterLab). Optionally, you could also include a longer description for the commit in the next text box.
+
+You can see a good example of an informative title and long description in this slide. The general guideline for the title is to summarize the purpose of the commit as a short descriptive message. It is recommended to write your messages in imperative mood (imagine that your commit should complete the sentence "this commit will...") and start with a capital letter. You can [read more about how to write a good commit message in this article](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/) and see a few examples below:
+
+- Bad messages:
+    - fixes an error
+    - added new code
+- Good messages:
+    - Reorder sections to make the README more welcoming
+    - Fix notebook error to allow for export with images
 
 ---
+
 ## JupyterLab: Making the commit
 
 
 <center>
 
-<img src='/module3/vc-commits-jupyter-5.png' width="90%" alt="404 image"/>
-
-<center/>
+<img src='/module3/vc-commits-jupyter-5.png' width="80%" alt="404 image"/>
 
+</center>
 
-Notes: **3. After recording a commit, the staging area should be empty.**
```

Comment:
Are you not keeping step 3?

Reply:
I removed step 3 here because I didn't think of it as something that we have to do when committing in JupyterLab. There are two steps/actions the student need to perform: Click the icon, and then add a commit message. The empty staging is more of a piece of information I think, but not associated with any action, so I added it as general text below instead of a step. But please feel free to add it as a third step if you disagree and prefer it that way when you record.